src/main/java/com/shopping_app/shoppingApp/service/UserService.java

```diff
@@ -0,0 +1,196 @@
+import com.shopping_app.shoppingApp.repository.CartRepository;
+import com.shopping_app.shoppingApp.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final AddressRepository addressRepository;
+    private final CartRepository cartRepository;
+    private final PasswordEncoder bCryptPasswordEncoder;
+    private final AuthenticationManager authenticationManager;
+    private final JwtTokenProvider tokenProvider;
+
+    public UserResponse registerUser(UserRegisterRequest userRegisterRequest) {
+        Optional<User> isUserExist = userRepository.findByEmail(userRegisterRequest.getEmail());
+        if (isUserExist.isPresent()) {
+            throw new UserAlreadyExist("User with this email Already exist", HttpStatus.BAD_REQUEST);
+        }
+        User user = convertToUserDomain(userRegisterRequest);
+        Set<Address> address = user.getAddress();
+        user.setAddress(null);
+        user.setRoleName(UserRole.CUSTOMER);
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+        User savedUser = userRepository.save(user);
+        setAddressToUser(address, savedUser);
+        // After registration create cart for users
+        createEmptyCart(savedUser);
+        savedUser.setAddress(address);
+        return convertToUserResponse(savedUser);
+    }
+
+    public UserLoginResponse loginUser(UserLoginRequest userLoginRequest) {
+        Authentication authentication = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(userLoginRequest.getEmail(), userLoginRequest.getPassword()));
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        String token = tokenProvider.createToken(userLoginRequest.getEmail());
+        return convertToUserLoginResponse(userLoginRequest, token);
+    }
+
+    public List<UserResponse> getAllUsers() {
+        List<User> users = userRepository.findAll();
+        return convertToUserDomainList(users);
+    }
+
+    public UserResponse getUserById(Long id) {
+        Optional<User> user = userRepository.findById(id);
```

Comment:
Access control checks are missing here, can you add it?

Reply:
done